devicetypes/zooz/zooz-zse44-temperature-humidity-xs-sensor.src/zooz-zse44-temperature-humidity-xs-sensor.groovy

```diff
@@ -0,0 +1,424 @@
+		log.warn "Syncing ${changes} Change(s)"
+	}
+
+	if (state.wakeUpInterval != wakeUpInterval) {
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalSet(seconds: wakeUpInterval, nodeid:zwaveHubNodeId))
+		cmds << secureCmd(zwave.wakeUpV2.wakeUpIntervalGet())
+	}
+
+	configParams.each { name, param ->
+		Integer storedVal = getStoredVal(name)
+		Integer settingVal = getSettingVal(name)
+		if (storedVal != settingVal) {
+			logDebug "Changing ${param.title}(#${param.num}) from ${storedVal} to ${settingVal}"
+			cmds << secureCmd(zwave.configurationV1.configurationSet(parameterNumber: param.num, size: param.size, scaledConfigurationValue: settingVal))
+			cmds << secureCmd(zwave.configurationV1.configurationGet(parameterNumber: param.num))
+		}
+	}
+	return cmds
+}
+
+def ping() {
+	logDebug "ping()"
+}
+
+String secureCmd(cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		return zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		return cmd.format()
+	}
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description, commandClassVersions)
+	if (cmd) {
+		zwaveEvent(cmd)
+	} else {
+		log.warn "Unable to parse: $description"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCmd = cmd.encapsulatedCommand(commandClassVersions)
+	if (encapsulatedCmd) {
+		zwaveEvent(encapsulatedCmd)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+	}
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpNotification cmd) {
+	logDebug "Device Woke Up..."
+	List<String> cmds = []
+
+	cmds += getRefreshCmds()
+	cmds += getConfigureCmds()
+
+	if (!cmds) {
+		cmds << secureCmd(zwave.batteryV1.batteryGet())
+	}
+
+	cmds << secureCmd(zwave.wakeUpV2.wakeUpNoMoreInformation())
+	sendHubCommand(cmds, 250)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.wakeupv2.WakeUpIntervalReport cmd) {
+	logDebug "Wake Up Interval = ${cmd.seconds} seconds"
+	state.wakeUpInterval = cmd.seconds
+	refreshSyncStatus()
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	Integer val = (cmd.batteryLevel == 0xFF ? 1 : cmd.batteryLevel)
+	if (val > 100) {
+		val = 100
+	}
+	logDebug "Battery is ${val}%"
+	sendEvent(name:"battery", value:val, unit:"%", isStateChange: true)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logDebug "${cmd}"
+	switch (cmd.notificationType) {
+		case temperatureAlarm.notificationType:
+			sendAlarmEvent(temperatureAlarm, cmd)
+			break
+		case humidityAlarm.notificationType:
+			sendAlarmEvent(humidityAlarm, cmd)
+			break
+		default:
+			logDebug "${cmd}"
+	}
+}
+
+void sendAlarmEvent(Map alarm, cmd) {
+	String value
+	if ((cmd.event == alarm.highEvent) || (cmd.event == alarm.lowEvent)) {
+		value = (cmd.event == alarm.highEvent) ? alarm.highValue : alarm.lowValue
+	} else if (cmd.event == alarm.normalEvent) {
+		value = alarm.normalValue
```

Comment:
why not restructure your map so you can just do `value = alarm[cmd.event]` ?

Reply:
I've refactored that section in the version I just committed.